src/jdmtool/main.py

```diff
@@ -104,18 +107,16 @@ def wrapper(dev: SkyboundDevice, *args, **kwargs):
         dev.before_read()
 
         # TODO: Figure out the actual meaning of the iid and the "unknown" value.
-        iid = dev.get_iid()
-        if iid == 0x01004100:
-            # 16MB WAAS card
-            print("Detected data card: 16MB WAAS")
-            dev.set_memory_layout(SkyboundDevice.MEMORY_LAYOUT_16MB)
-        elif iid == 0x0100ad00:
-            # 4MB non-WAAS card
-            print("Detected data card: 4MB non-WAAS")
-            dev.set_memory_layout(SkyboundDevice.MEMORY_LAYOUT_4MB)
+        card = dev.get_card()
+
+        log.debug(f"Detected card: {card}")
```

Comment:
When logging, I think you're supposed to use `%s` and extra args instead of f-strings, i.e.:
```suggestion
        log.debug("Detected card: %s", card)
```

The idea being, it only evaluates everything if logging is enabled.

Also, in this case, it's pretty much identical to the `print` below...

Reply:
Yup, i get it.  Was favoring readability over performance optimization in this particular case.  I'll remove entirely.